engine/runner/src/main/java/org/enso/runner/Main.java

```diff
@@ -1294,9 +1306,78 @@ private void println(String msg) {
     System.out.println(msg);
   }
 
-  private void launch(String[] args) {
+  private void launch(String[] args) throws IOException, InterruptedException, URISyntaxException {
     var options = buildOptions();
     var line = preprocessArguments(options, args);
+
+    if (line.hasOption(JVM_OPTION)) {
+      var jvm = line.getOptionValue(JVM_OPTION);
+      var current = System.getProperty("java.home");
+      if (current == null || !current.equals(jvm)) {
+        var loc = Main.class.getProtectionDomain().getCodeSource().getLocation();
+        var commandAndArgs = new ArrayList<String>();
+        JVM_FOUND:
+        if (jvm == null) {
+          var env = new Environment() {};
+          var dm = new DistributionManager(env);
+          var paths = dm.paths();
+          var files = paths.runtimes().toFile().listFiles();
```

Comment:
There is [RuntimeVersionManager](https://github.com/enso-org/enso/blob/2638ba8d8e170ee3bdd83a34263c12020adfdced/lib/scala/runtime-version-manager/src/main/scala/org/enso/runtimeversionmanager/components/RuntimeVersionManager.scala#L41) but I have no idea how shall I get access to its instance? I am glad I managed to instantiate `DistributionManager`... CCing @radeusgd, @4e6 

But now one can run `./bin/enso --jvm --run x.enso` and one of the installed GraalVM's is found and used.

Reply:
> There is [RuntimeVersionManager](https://github.com/enso-org/enso/blob/2638ba8d8e170ee3bdd83a34263c12020adfdced/lib/scala/runtime-version-manager/src/main/scala/org/enso/runtimeversionmanager/components/RuntimeVersionManager.scala#L41) but I have no idea how shall I get access its instance?

You can probably use some of the helpers in `DefaultManagers`.

Reply:
Note however that 90% of `RuntimeVersionManager` logic is downloading and installing a missing GraalVM runtime associated with the current engine. 

(Which runtime the engine 'wants' is specified in the manifest file)
![image](https://github.com/enso-org/enso/assets/1436948/caddcbb6-4bc8-4353-ad1f-0f88b9a05c2d)

If you **don't want** the downloading mechanism, then most of the facilities of the `RuntimeVersionManager` won't be useful to you - initialization of it is so complex because it covers things like displaying progress to the user (via CLI progress bar, or via Language Server messages etc.) or managing locks to avoid 2 concurrently running launchers overwriting each other's data and leaving a corrupted GraalVM installation. If you are not installing new JVMs, but only want to detect existing ones, the `DistributionManager` already does most of the work. I guess you'd just want to extract the `findGraalRuntime` part. Everything else is exactly for handling downloading/installation.